include/deal.II/differentiation/ad/ad_drivers.h

```diff
@@ -275,6 +275,81 @@ namespace Differentiation
       void
       activate_tape(const typename Types<ADNumberType>::tape_index tape_index);
 
+      /**
+       * Return a flag that, when <code>true</code>, indicates that the retaping
+       * of the dependent function for the chosen @p tape_index is necessary for
+       * a reliable computation to be performed.
+       * This may be necessary if a sign comparison within branched operations
+       * yields different results to those computed at the original tape
+       * evaluation point.
+       *
+       * This issue, known as "branch switching", can be clarified by means of
+       * a trivial example:
+       * @code
+       * ADNumberType func (ADNumberType x, ADNumberType y, ADNumberType z)
+       * {
+       *   if (x < y)
+       *     return y;
+       *   else
+       *     return x*z;
+       * }
```

Comment:
Thanks for adding this example! 

In order for the system to determine that retaping is necessary, I guess it doesn't just record the parse tree of the operations applied, but also that at one point it checked the condition `x<y`? Without remembering this condition, it would not be able to tell that retaping is necessary.

If I'm correct with this, could you add a sentence to this effect to the end of the paragraph following this example?